File: app/Models/Evenement.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Evenement extends Model
{
    use HasFactory;

    protected $fillable = [
        'titre',
        'description',
        'date_debut',
        'date_fin',
        'lieu',
        'type'
    ];

    protected $casts = [
        'date_debut' => 'datetime',
        'date_fin' => 'datetime',
    ];

    public function classes()
    {
        return $this->belongsToMany(Classe::class, 'classe_evenement');
    }
}

File: professeur_frontend/lib/models/evenement.dart
class Evenement {
  final int id;
  final String titre;
  final String? description;
  final DateTime dateDebut;
  final DateTime dateFin;
  final String? lieu;
  final String type;

  Evenement({
    required this.id,
    required this.titre,
    this.description,
    required this.dateDebut,
    required this.dateFin,
    this.lieu,
    required this.type,
  });

  factory Evenement.fromJson(Map<String, dynamic> json) {
    return Evenement(
      id: json['id'],
      titre: json['titre'],
      description: json['description'],
      dateDebut: DateTime.parse(json['date_debut']),
      dateFin: DateTime.parse(json['date_fin']),
      lieu: json['lieu'],
      type: json['type'],
    );
  }
}
